### PR TITLE
Remove travis.sh/docker.sh and update development docs regarding the CI.

### LINKED ISSUE #1723: Remove travis.sh
We're using GHA now.

### PR DESCRIPTION
**Pull request by ni4**
Fixes #1723

### CODE REVIEW

File: docs/develop.adoc [hunk 1]
```diff
@@ -74,36 +74,33 @@ history of `master`, depending on how they are merged.
...
-Sometimes tests fail in Travis CI and you will want to reproduce them
-locally for easier troubleshooting.
-
-We can use a container for this, like so:
+If tests fail in CI, you may attempt to reproduce those locally via `ctest` command:
 
 [source,console]
 --
-./travis.sh
+`ctest -j4 -V -R rnp_tests`
```
Comment (ronaldtse) [inline docs/develop.adoc:92]: Could remove the surrounding backticks.

File: docs/develop.adoc [hunk 2]
```diff
@@ -74,36 +74,33 @@ history of `master`, depending on how they are merged.
...
 --
 
-Inside the container, you can do local CI runs like so:
+Or, more specific:
 
 [source,console]
 --
-env GPG_VERSION=beta BUILD_MODE=sanitize ci/local.sh
+`ctest -V -R cli_tests-Misc`
```
Comment (ronaldtse) [inline docs/develop.adoc:99]: Could remove the surrounding backticks.

**Review by ronaldtse — CHANGES_REQUESTED** [review]

File: docs/develop.adoc (hunk 1 above)
Comment (ni4) [reply docs/develop.adoc:92]: Thanks, will fix (as well as one below).

**Review by ronaldtse — APPROVED** [review]

**Comment by ronaldtse** [comment]
Thanks @ni4 !

**Pull request merged**